File: example/sema-java-demo/src/main/java/com/semademo/BridgeEvent.java
package com.semademo;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;

/**
 * Node.js → 宿主 事件帧
 */
public class BridgeEvent {

    @JsonProperty("event")
    public String event;

    @JsonProperty("data")
    public JsonNode data;

    @JsonProperty("cmdId")
    public String cmdId;

    @JsonProperty("error")
    public String error;
}

File: example/sema-bridge-clients/sema-python-demo/bridge_event.py
from dataclasses import dataclass
from typing import Any, Optional


@dataclass
class BridgeEvent:
    """Node.js → 宿主 事件帧"""

    event: str = ""
    data: Any = None
    cmd_id: Optional[str] = None
    error: Optional[str] = None

    @classmethod
    def from_dict(cls, d: dict) -> "BridgeEvent":
        return cls(
            event=d.get("event", ""),
            data=d.get("data"),
            cmd_id=d.get("cmdId"),
            error=d.get("error"),
        )
